### PR TITLE
Automatic update of translations

### PR DESCRIPTION
**Pull request by jonasfranz**
This PR adds a build step for master-build. The step uploads `locale_en-US.ini` to crowdin via the provided API.

@tboerger This PR needs a new drone secret `crowdin_key` containing the API key of the gitea project. (https://crowdin.com/project/gitea/settings#api)

@tboerger Is the english source file called "locale_en-US.ini" at crowdin?

**EDIT**
> As said on discord, maybe it makes more sense to build a drone plugin for upload and download of translations, this way anybody can prepare a malicious pull request to read the access token because I can only limit the injection to the used image and the events.

I've created a plugin for this PR and updated the .drone.yml to use this plugin. The plugin requires the crowdin key as secret `CROWDIN_KEY` (CC @tboerger).

Link to plugin repo: https://github.com/JonasFranzDEV/drone-crowdin
Published on Docker hub: https://hub.docker.com/r/jonasfranz/crowdin/

Working example of the plugin:

Drone: https://drone.jonasfranz.software/JonasFranzDEV/Crowdin-Demo/5
Git: https://git.jonasfranz.software/JonasFranzDEV/Crowdin-Demo
Crowdin: https://crowdin.com/project/gitea-demo/activity_stream

### CODE REVIEW

**Comment by lunny** [comment]
I have added secret for crowdin_key. locale_en-US.ini is the right name.

**Comment by tboerger** [comment]
As said on discord, maybe it makes more sense to build a drone plugin for upload and download of translations, this way anybody can prepare a malicious pull request to read the access token because I can only limit the injection to the used image and the events.

**Comment by jonasfranz** [comment]
> As said on discord, maybe it makes more sense to build a drone plugin for upload and download of translations, this way anybody can prepare a malicious pull request to read the access token because I can only limit the injection to the used image and the events.

I've created a plugin for this PR and updated the .drone.yml to use this plugin. The plugin requires the crowdin key as secret `CROWDIN_KEY` (CC @tboerger).

Link to plugin repo: https://github.com/JonasFranzDEV/drone-crowdin
Published on Docker hub: https://hub.docker.com/r/jonasfranz/crowdin/

**Comment by jonasfranz** [comment]
Working example of the plugin:

Drone: https://drone.jonasfranz.software/JonasFranzDEV/Crowdin-Demo/5
Git: https://git.jonasfranz.software/JonasFranzDEV/Crowdin-Demo
Crowdin: https://crowdin.com/project/gitea-demo/activity_stream

File: .drone.yml [hunk 1]
```diff
@@ -207,6 +207,16 @@ pipeline:
       event: [ push ]
       branch: [ master ]
 
+  translations:
+    image: jonasfranz/crowdin
+    secrets: [ CROWDIN_KEY ]
```
Comment (lafriks) [inline .drone.yml]: Why secret is uppercase while others in this file are lowercase?

**Comment by lafriks** [comment]
LGTM you should probably submit this plugin to drone-plugins :)

**Review by lafriks — APPROVED** [review]

**Comment by JonasFranzDEV** [comment]
@lafriks I do not now how to submit the plugin to drone-plugins xD

**Comment by jonasfranz** [comment]
@lafriks I created a PR for the plugin: https://github.com/drone/drone-plugin-index/pull/71

File: .drone.yml [hunk 2]
```diff
@@ -207,6 +207,16 @@ pipeline:
       event: [ push ]
       branch: [ master ]
 
+  translations:
+    image: jonasfranz/crowdin
```
Comment (tboerger) [inline .drone.yml:211]: pull: true below this line

Comment (jonasfranz) [reply .drone.yml:211]: @tboerger Done

**Comment by daviian** [comment]
As I understand the plugin only uploads specified files to crowdin to make them available for translation..
But shouldn't it also download existing translations for release?

**Comment by jonasfranz** [comment]
@daviian Yes it should, but actually it only supports uploading files. I could add downloading in future. But I think that it is enough for this PR.

**Comment by daviian** [comment]
LGTM

**Comment by tboerger** [comment]
Maybe I can start to integrate that together with Franz when this plugin gets transferred to the drone-plugins org

**Comment by lunny** [comment]
drone is failed, see https://drone.gitea.io/go-gitea/gitea/910/15

**Comment by jonasfranz** [comment]
We have to add ignore_branch since we are not using branches at crowdin

**Comment by jonasfranz** [comment]
@lunny

**Comment by lunny** [comment]
@JonasFranzDEV OK. So should I change some settings on crowdin or could you send another PR to fix that?

**Comment by lafriks** [comment]
PR to fix that #2599

**Comment by tboerger** [comment]
A note: currently this plugin doesn't commit the changes to the repository, maybe we have to add another step based on https://github.com/appleboy/drone-git-push to get these changes commited with a message including `[skip ci]` to avoid another CI run?

**Comment by lafriks** [comment]
Downloading changes from crowdin and pushing to git would be best scenario!

**Comment by jonasfranz** [comment]
@tboerger @lafriks I've added a section in the plugin documentation which describes the git push variant: https://github.com/JonasFranzDEV/drone-crowdin/blob/master/DOCS.md#commit-changes

**Pull request merged**